lib/Memcached.php

```diff
@@ -47,8 +47,21 @@ public function get($key, $default = null)
             throw new InvalidArgumentException('$key must be a string');
         }
         $result = $this->memcached->get($key);
-        if (false === $result && \Memcached::RES_NOTFOUND === $this->memcached->getResultCode()) {
-            return $default;
+        if (false === $result) {
+            // Note: result can be false because the key exists and has the value boolean false,
+            //       or because something went wrong (cache miss, memcached server timeout...)
+            // So we need to check the result code to work out what to do.
+            $resultCode = $this->memcached->getResultCode();
+            if (\Memcached::RES_BAD_KEY_PROVIDED === $resultCode) {
+                // The key was a string but is invalid. Maybe it was too long, contained whitespace etc.
+                throw new InvalidArgumentException('$key was not valid');
```

Comment:
this looks like a `\InvalidArgumentException` but not a `\Psr\SimpleCache\InvalidArgumentException::class`

Reply:
This and line 47 are `Sabre\Cache\InvalidArgumentException` and that is:
```
class InvalidArgumentException extends \InvalidArgumentException implements \Psr\SimpleCache\InvalidArgumentException
{
}
```

So it is both `\InvalidArgumentException` and `\Psr\SimpleCache\InvalidArgumentException`

That is existing code. Does that seem correct/reasonable?